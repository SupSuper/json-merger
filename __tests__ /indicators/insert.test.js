const jsonMerger = require("../../dist");
const {testConfig} = require("../__helpers__");

describe("when merging two arrays and the source item has", function () {

    test("$insert: 0 it should prepend the source array item", function () {

        const object1 = {
            "a": [
                {
                    "aa": "string"
                }
            ]
        };

        const object2 = {
            "a": [
                {
                    "$insert": 0,
                    "bb": "prepend"
                }
            ]
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });

    test("$insert: 1 it should insert the source array item at index 1", function () {

        const object1 = {
            "a": [
                {
                    "aa1": "string"
                },
                {
                    "aa2": "string"
                }
            ]
        };

        const object2 = {
            "a": [
                {
                    "$insert": 1,
                    "bb": "insert at index 1"
                }
            ]
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });

    test("$insert: 1 and an other item also has $insert: 1 it should insert the last item at index 1", function () {

        const object1 = {
            "a": [
                {
                    "aa1": "string"
                },
                {
                    "aa2": "string"
                }
            ]
        };

        const object2 = {
            "a": [
                {
                    "$insert": 1,
                    "bb": "insert at index 2"
                },
                {
                    "$insert": 1,
                    "cc": "insert at index 1"
                }
            ]
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });

    test("$insert: -1 it should insert the source array item before the last item", function () {

        const object1 = {
            "a": [
                {
                    "aa1": "string"
                },
                {
                    "aa2": "string"
                }
            ]
        };

        const object2 = {
            "a": [
                {
                    "$insert": -1,
                    "bb": "insert before last"
                }
            ]
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });

    test("$insert: - it should append the source array item", function () {

        const object1 = {
            "a": [
                {
                    "aa1": "string"
                },
                {
                    "aa2": "string"
                }
            ]
        };

        const object2 = {
            "a": [
                {
                    "$insert": "-",
                    "bb": "append"
                }
            ]
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });

    test("an $insert value higher than the target array length it should append the array item", function () {

        const object1 = {
            "a": [
                {
                    "aa1": "string"
                },
                {
                    "aa2": "string"
                }
            ]
        };

        const object2 = {
            "a": [
                {
                    "$insert": 10,
                    "bb": "append"
                }
            ]
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });
});

test("$insert on a non array item should do nothing and be stripped", function () {

    const object1 = {
        "1a": {
            "1aa": 1
        }
    };

    const object2 = {
        "2b": {
            "$insert": true,
            "2bb": 2
        }
    };

    const result = jsonMerger.mergeObjects([object1, object2], testConfig());

    expect(result).toMatchSnapshot();
});
