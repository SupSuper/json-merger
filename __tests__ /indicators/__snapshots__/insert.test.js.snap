// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`$insert on a non array item should do nothing and be stripped 1`] = `
"{
	\\"1a\\": {
		\\"1aa\\": 1
	},
	\\"2b\\": {
		\\"2bb\\": 2
	}
}"
`;

exports[`when merging two arrays and the source item has $insert: - it should append the source array item 1`] = `
"{
	\\"a\\": [
		{
			\\"aa1\\": \\"string\\"
		},
		{
			\\"aa2\\": \\"string\\"
		},
		{
			\\"bb\\": \\"append\\"
		}
	]
}"
`;

exports[`when merging two arrays and the source item has $insert: -1 it should insert the source array item before the last item 1`] = `
"{
	\\"a\\": [
		{
			\\"aa1\\": \\"string\\"
		},
		{
			\\"bb\\": \\"insert before last\\"
		},
		{
			\\"aa2\\": \\"string\\"
		}
	]
}"
`;

exports[`when merging two arrays and the source item has $insert: 0 it should prepend the source array item 1`] = `
"{
	\\"a\\": [
		{
			\\"bb\\": \\"prepend\\"
		},
		{
			\\"aa\\": \\"string\\"
		}
	]
}"
`;

exports[`when merging two arrays and the source item has $insert: 1 and an other item also has $insert: 1 it should insert the last item at index 1 1`] = `
"{
	\\"a\\": [
		{
			\\"aa1\\": \\"string\\"
		},
		{
			\\"cc\\": \\"insert at index 1\\"
		},
		{
			\\"bb\\": \\"insert at index 2\\"
		},
		{
			\\"aa2\\": \\"string\\"
		}
	]
}"
`;

exports[`when merging two arrays and the source item has $insert: 1 it should insert the source array item at index 1 1`] = `
"{
	\\"a\\": [
		{
			\\"aa1\\": \\"string\\"
		},
		{
			\\"bb\\": \\"insert at index 1\\"
		},
		{
			\\"aa2\\": \\"string\\"
		}
	]
}"
`;

exports[`when merging two arrays and the source item has an $insert value higher than the target array length it should append the array item 1`] = `
"{
	\\"a\\": [
		{
			\\"aa1\\": \\"string\\"
		},
		{
			\\"aa2\\": \\"string\\"
		},
		{
			\\"bb\\": \\"append\\"
		}
	]
}"
`;
